crates/router/src/services/authentication.rs

```diff
@@ -1980,62 +1984,82 @@ where
     default_auth
 }
 
+#[derive(Clone)]
 #[cfg(feature = "recon")]
-pub struct ReconAdmin;
+pub struct UserFromTokenWithAuthData(pub UserFromToken, pub AuthenticationDataWithUser);
```

Comment:
`UserFromToken` should be enough right?

Reply:
Added based on @jarnura's suggestions to fetch user in auth layer